src/audio/volume.c

```diff
@@ -250,6 +264,61 @@ static inline void volume_set_chan(struct comp_dev *dev, int chan, uint32_t vol)
 		v = VOL_MAX;
 
 	cd->tvolume[chan] = v;
+
+	/* Check ramp type */
+	switch (pga->ramp) {
+	case SOF_VOLUME_LINEAR:
+
+		/* Assume the ramp length (initial_ramp [ms]) describes
+		 * time of mute to 0 dB ramp. The actual volume scale min/max
+		 * is not known. If initial_ramp is zero (or negative) use
+		 * step size that reaches 0 dB gain in one step (VOL_ZERO_DB).
+		 */
+		if (pga->initial_ramp > 0)
+			inc = MAX(VOL_RAMP_STEP_CONST / pga->initial_ramp, 1);
+		else
+			inc = VOL_ZERO_DB;
+
+		delta = cd->tvolume[chan] - cd->volume[chan];
+		if (delta < 0)
+			delta_abs = -delta;
+		else
+			delta_abs = delta;
+
+		trace_volume("chan = %d", chan);
+		trace_volume("delta = %d", delta);
+		trace_volume("delta_abs = %d", delta_abs);
```

Comment:
Can be done in one trace message.

Reply:
These are something to remove. I was just checking that the code is sane with these.

Reply:
These would be removed, or maybe converted to tracev().